Lib/test/test_clinic.py

```diff
@@ -3030,5 +3030,97 @@ def test_suffix_all_lines(self):
         self.assertEqual(out, expected)
 
 
+class ClinicReprTests(unittest.TestCase):
+    def test_Block_repr(self):
+        block = clinic.Block("foo")
+        expected_repr = "<clinic.Block 'text' input='foo' output=None>"
+        self.assertEqual(repr(block), expected_repr)
+
+        block2 = clinic.Block("bar", "baz", [], "eggs", "spam")
+        expected_repr_2 = "<clinic.Block 'baz' input='bar' output='eggs'>"
+        self.assertEqual(repr(block2), expected_repr_2)
+
+        block3 = clinic.Block(
+            input="longboi_" * 100,
+            dsl_name="wow_so_long",
+            signatures=[],
+            output="very_long_" * 100,
+            indent=""
+        )
+        expected_repr_3 = (
+            "<clinic.Block 'wow_so_long' input='longboi_longboi_longboi_l...' output='very_long_very_long_very_...'>"
+        )
+        self.assertEqual(repr(block3), expected_repr_3)
+
+    def test_Destination_repr(self):
+        destination = clinic.Destination(
+            "foo", type="file", clinic=FakeClinic(), args=("eggs",)
+        )
+        self.assertEqual(
+            repr(destination), "<clinic.Destination 'foo' type='file' file='eggs'>"
+        )
+
+        destination2 = clinic.Destination("bar", type="buffer", clinic=FakeClinic())
+        self.assertEqual(repr(destination2), "<clinic.Destination 'bar' type='buffer'>")
+
+    def test_Module_repr(self):
+        module = clinic.Module("foo", FakeClinic())
+        self.assertRegex(repr(module), r"<clinic.Module 'foo' at [a-zA-Z0-9]+>")
```

Comment:
It looks to me we can match with `\w` (alphanum) instead of `[a-zA-Z0-9]`? Ditto for the others.

Reply:
`[a-zA-Z0-9]` is slightly more precise:

```pycon
>>> import re
>>> re.search(r"[a-zA-Z0-9]", "_") is None
True
>>> re.search(r"\w", "_")
<re.Match object; span=(0, 1), match='_'>
>>> re.search(r"[a-zA-Z0-9]", "â") is None
True
>>> re.search(r"\w", "â")
<re.Match object; span=(0, 1), match='â'>
```

But maybe it doesn't matter too much?

We could also make `\w` match only ASCII characters by using the `(?a)` inline flag:

```pycon
>>> re.search(r"(?a)\w", "â") is None
True
```

But I prefer not to use the more obscure features of Python's regular-expression syntax if I can help it 😄. It also still matches `_` even if you have the inline `(?a)` flag, since `_` is an ASCII character:

```pycon
>>> re.search(r"(?a)\w", "_")
<re.Match object; span=(0, 1), match='_'>
```

Reply:
...But the code is just using `str(id(self))`, so it's not a hex string at all. We can just use `\d`. Oops.